src/lsstseries/analysis/structure_function_calculators/basic_structure_function_calculator.py

```diff
@@ -0,0 +1,126 @@
+            f_mask = np.isnan(lc_fluxes)
+            e_mask = np.isnan(lc_errors)  # always mask out nan errors?
+            lc_mask = np.logical_or(t_mask, f_mask, e_mask)
+
+            lc_times = lc_times[~lc_mask]
+            lc_fluxes = lc_fluxes[~lc_mask]
+            lc_errors = lc_errors[~lc_mask]
+
+            # compute d_times (difference of times) and
+            # d_fluxes (difference of magnitudes, i.e., fluxes) for all gaps
+            # d_times - difference of times
+            dt_matrix = lc_times.reshape((1, lc_times.size)) - lc_times.reshape((lc_times.size, 1))
+            d_times = dt_matrix[dt_matrix > 0].flatten()
+
+            # d_fluxes - difference of fluxes
+            df_matrix = lc_fluxes.reshape((1, lc_fluxes.size)) - lc_fluxes.reshape((lc_fluxes.size, 1))
+            d_fluxes = df_matrix[dt_matrix > 0].flatten()
+
+            # err^2 - errors squared
+            err2_matrix = (
+                lc_errors.reshape((1, lc_errors.size)) ** 2 + lc_errors.reshape((lc_errors.size, 1)) ** 2
+            )
+            err2s = err2_matrix[dt_matrix > 0].flatten()
+
+            # corrected each pair of observations
+            cor_flux2 = d_fluxes**2 - err2s
+
+            # build stack of times and fluxes
+            self._dts.append(d_times)
+            cor_flux2_all.append(cor_flux2)
+
+        # combining treats all lightcurves as one when calculating the structure function
+        if self._argument_container.combine and len(self._time) > 1:
+            self._dts = np.hstack(np.array(self._dts, dtype="object"))
+            cor_flux2_all = np.hstack(np.array(cor_flux2_all, dtype="object"))
+
+            # binning
+            if self._bins is None:
+                self._bin_dts(self._dts)
+
+            # structure function at specific dt
+            # the line below will throw error if the bins are not covering the whole range
+            sfs, bin_edgs, _ = binned_statistic(self._dts, cor_flux2_all, "mean", self._bins)
+            return [(bin_edgs[0:-1] + bin_edgs[1:]) / 2], [sfs]
+        # Not combining calculates structure function for each light curve independently
+        else:
+            # may want to raise warning if len(times) <=1 and combine was set true
+            sfs_all = []
+            t_all = []
+            for lc_idx in range(len(self._dts)):
+                if len(self._dts[lc_idx]) > 1:
+                    # binning
+                    self._bin_dts(self._dts[lc_idx])
+                    sfs, bin_edgs, _ = binned_statistic(
+                        self._dts[lc_idx], cor_flux2_all[lc_idx], "mean", self._bins
+                    )
+                    sfs_all.append(sfs)
+                    t_all.append((bin_edgs[0:-1] + bin_edgs[1:]) / 2)
+                else:
+                    sfs_all.append(np.array([]))
+                    t_all.append(np.array([]))
+            return t_all, sfs_all
+
+    @staticmethod
+    def name_id():
```

Comment:
Pylint should be ok with inherited methods not having docstrings (and I think it would get more angry about the repeated lines this will generate).